src/evict/evict_lru.c

```diff
@@ -786,11 +786,11 @@ __evict_pass(WT_SESSION_IMPL *session)
         if (eviction_progress == cache->eviction_progress) {
             if (WT_CLOCKDIFF_MS(time_now, time_prev) >= 20 && F_ISSET(cache, WT_CACHE_EVICT_HARD)) {
                 if (cache->evict_aggressive_score < WT_EVICT_SCORE_MAX)
-                    ++cache->evict_aggressive_score;
+                    (void)__wt_atomic_addv32(&cache->evict_aggressive_score, 1);
```

Comment:
These don't need the volatile cast, which means you can use __wt_atomic_load32() instead.

Reply:
Volatile is used in WiredTiger to prevent compiler optimisation with respect a given variable. Generally we aim to avoid using volatile favouring WT_READ_ONCE or WT_WRITE_ONCE. These primitives are better defined here https://www.kernel.org/doc/Documentation/memory-barriers.txt see the heading:
```
COMPILER BARRIER
----------------
```

The current atomic macros in WiredTiger have two variants `volatile` and non `volatile`. We currently use the volatile implementation if the variable is already declared as volatile. In this case that variable isn't. Note that the atomic macros are old and likely need review but for now that is how we use them.

Reply:
I learned, thanks.
:)